server/protocols/webadmin/webadmin-rabbitmq/src/test/java/org/apache/james/webadmin/routes/RabbitMQMailQueuesRoutesTest.java

```diff
@@ -0,0 +1,144 @@
+                new RabbitMQMailQueuesRoutes(mailQueueFactory, clock, jsonTransformer, taskManager),
+                new TasksRoutes(taskManager, jsonTransformer, DTOConverter.of()))
+            .start();
+    }
+
+    RequestSpecification buildRequestSpecification(WebAdminServer server) {
+        return new RequestSpecBuilder()
+            .setContentType(ContentType.JSON)
+            .setAccept(ContentType.JSON)
+            .setBasePath("/")
+            .setPort(server.getPort().getValue())
+            .setConfig(newConfig().encoderConfig(encoderConfig().defaultContentCharset(StandardCharsets.UTF_8)))
+            .build();
+    }
+
+    @BeforeEach
+    void setUp() {
+        mailQueueFactory = mock(RabbitMQMailQueueFactory.class);
+        webAdminServer = createServer(mailQueueFactory);
+        RestAssured.requestSpecification = buildRequestSpecification(webAdminServer);
+        RestAssured.enableLoggingOfRequestAndResponseIfValidationFails();
+    }
+
+    @AfterEach
+    void tearDown() {
+        webAdminServer.destroy();
+    }
+
+    @Test
+    void triggeringARepublishNotProcessedMailsShouldCreateATask() {
+        when(mailQueueFactory.getQueue(any())).thenReturn(Optional.of(mock(RabbitMQMailQueue.class)));
+        given()
+            .queryParam("action", "RepublishNotProcessedMails")
+            .queryParam("olderThan", "1d")
+        .when()
+            .post(MailQueueRoutes.BASE_URL + "/spooler")
+        .then()
+            .statusCode(HttpStatus.CREATED_201);
+
+        given()
+        .when()
+            .get("/tasks")
+        .then()
+            .statusCode(HttpStatus.OK_200)
+            .body("", hasSize(1));
+    }
+
+    @Test
+    void triggeringARepublishNotProcessedMailsWhenTheQueueHasNotBeenInitializedShouldFail() {
+        when(mailQueueFactory.getQueue(any())).thenReturn(Optional.empty());
+        given()
+            .queryParam("action", "RepublishNotProcessedMails")
+            .queryParam("olderThan", "1d")
+        .when()
+            .post(MailQueueRoutes.BASE_URL + "/spooler")
+        .then()
+            .statusCode(HttpStatus.NOT_FOUND_404)
+            .body("message", containsString("MailQueueName{value=spooler} can not be found"));
+    }
+
+    @Test
+    void triggeringARepublishNotProcessedMailsWithAnInvalidOlderThanShouldFail() {
+        when(mailQueueFactory.getQueue(any())).thenReturn(Optional.of(mock(RabbitMQMailQueue.class)));
+        given()
+            .queryParam("action", "RepublishNotProcessedMails")
+            .queryParam("olderThan", "invalidValue")
+        .when()
+            .post(MailQueueRoutes.BASE_URL + "/spooler")
+        .then()
+            .statusCode(HttpStatus.BAD_REQUEST_400)
+            .body("message", containsString("Invalid olderThan"))
+            .body("details", containsString("Supplied value do not follow the unit format (number optionally suffixed with a string representing the unit"));
```

Comment:
do you miss some tests about the olderThan parameter? (At least checking that passing a number gives some days, and passing a number + a unit gives the good unit)

Reply:
Seems hard to test, we do not display it in the details, only the absolute time (which is more ergonomic).